src/lsp/common.ts

```diff
@@ -0,0 +1,53 @@
+import * as vscode from "vscode";
+import { LanguageServerClient } from ".";
+import { LanguageServer } from "../types";
+import { SwiftMesonLspLanguageClient } from "./swift-mesonlsp";
+import { Uri } from "vscode";
+
+export async function createLanguageServerClient(server: LanguageServer, download: boolean, context: vscode.ExtensionContext): Promise<LanguageServerClient | null> {
```

Comment:
I think this might make sense to factor into the LanguageServerClient class.

```typescript
public static async LanguageServerClient.create(...): Promise<LanguageServerClient | null>;
```

Reply:
Sadly moving it does not work:
```
2023-08-08 18:45:28.158 [error] TypeError: Class extends value undefined is not a constructor or null
	at Object.<anonymous> (/home/user/Projects/vscode-meson/out/lsp/swift-mesonlsp.js:6:49)
	at Module._compile (node:internal/modules/cjs/loader:1141:14)
	at Module._extensions..js (node:internal/modules/cjs/loader:1196:10)
	at Module.load (node:internal/modules/cjs/loader:1011:32)
	at Module._load (node:internal/modules/cjs/loader:846:12)
	at f._load (node:electron/js2c/asar_bundle:2:13330)
	at o._load (/usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:130:34104)
	at f._load (/usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:130:30479)
	at I._load (/usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:24270)
	at Module.require (node:internal/modules/cjs/loader:1035:19)
	at require (node:internal/modules/cjs/helpers:102:18)
	at Object.<anonymous> (/home/user/Projects/vscode-meson/out/lsp/index.js:15:26)
	at Module._compile (node:internal/modules/cjs/loader:1141:14)
	at Module._extensions..js (node:internal/modules/cjs/loader:1196:10)
	at Module.load (node:internal/modules/cjs/loader:1011:32)
	at Module._load (node:internal/modules/cjs/loader:846:12)
	at f._load (node:electron/js2c/asar_bundle:2:13330)
	at o._load (/usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:130:34104)
	at f._load (/usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:130:30479)
	at I._load (/usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:24270)
	at Module.require (node:internal/modules/cjs/loader:1035:19)
	at require (node:internal/modules/cjs/helpers:102:18)
	at Object.<anonymous> (/home/user/Projects/vscode-meson/out/lsp/common.js:5:12)
	at Module._compile (node:internal/modules/cjs/loader:1141:14)
	at Module._extensions..js (node:internal/modules/cjs/loader:1196:10)
	at Module.load (node:internal/modules/cjs/loader:1011:32)
	at Module._load (node:internal/modules/cjs/loader:846:12)
	at f._load (node:electron/js2c/asar_bundle:2:13330)
	at o._load (/usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:130:34104)
	at f._load (/usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:130:30479)
	at I._load (/usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:24270)
	at Module.require (node:internal/modules/cjs/loader:1035:19)
	at require (node:internal/modules/cjs/helpers:102:18)
	at Object.<anonymous> (/home/user/Projects/vscode-meson/out/extension.js:13:18)
	at Module._compile (node:internal/modules/cjs/loader:1141:14)
	at Module._extensions..js (node:internal/modules/cjs/loader:1196:10)
	at Module.load (node:internal/modules/cjs/loader:1011:32)
	at Module._load (node:internal/modules/cjs/loader:846:12)
	at f._load (node:electron/js2c/asar_bundle:2:13330)
	at o._load (/usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:130:34104)
	at f._load (/usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:130:30479)
	at I._load (/usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:24270)
	at Module.require (node:internal/modules/cjs/loader:1035:19)
	at require (node:internal/modules/cjs/helpers:102:18)
	at Function.i [as __$__nodeRequire] (/usr/share/codium/resources/app/out/vs/loader.js:5:98)
	at r._loadCommonJSModule (/usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:130:32106)
	at async Promise.all (index 0)
```
That was the reason in the beginning I moved it into the common.ts. The explanation according to SO was some cyclic import/dependency cycle or something like that.

Reply:
I'll pull the latest branch this week to try it for myself just to see what I can dig up